packages/@vue/cli/lib/util/inferRootOptions.js

```diff
@@ -8,11 +8,6 @@ module.exports = function inferRootOptions (pkg) {
   // projectName
   rootOptions.projectName = pkg.name
 
-  // router
-  if ('vue-router' in deps) {
-    rootOptions.router = true
```

Comment:
We don't have to remove this. Router is now a generator-only plugin anyway. Checking for `vue-router` is fine and we need to support the legacy rootOptions for now.